public/app/features/dashboard/utils/dashboard.ts

```diff
@@ -0,0 +1,91 @@
+import { chain, cloneDeep, defaults, find } from 'lodash';
+
+import { PanelPluginMeta } from '@grafana/data';
+import config from 'app/core/config';
+import { LS_PANEL_COPY_KEY } from 'app/core/constants';
+import store from 'app/core/store';
+import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
+import { calculateNewPanelGridPos } from 'app/features/dashboard/utils/panel';
+
+export const onCreateNewPanel = (dashboard: DashboardModel): number | undefined => {
+  const newPanel: Partial<PanelModel> = {
+    type: 'timeseries',
+    title: 'Panel Title',
+    gridPos: calculateNewPanelGridPos(dashboard),
+  };
+
+  dashboard.addPanel(newPanel);
+  return newPanel.id;
+};
+
+export const onCreateNewRow = (dashboard: DashboardModel) => {
```

Comment:
```suggestion
export function onCreateNewRow(dashboard: DashboardModel) {
```
nit pick, try to use normal functions instead of variables assigned with arrow function, easier to read in my mind and easier to see it's a function.